### ISSUE TITLE
CRC32 checksum failed extracting ZIP with specific PDF

### ISSUE THREAD

**Issue by EthraZa** [cthackers/adm-zip#330]
The CRC32 error always happen if I include this PDF inside the ZIP, does no matter if it's alone inside the ZIP or not. It happens with others PDFs too, but not all of them, I just got this one.
[adm-zip-crc32error.zip](https://github.com/cthackers/adm-zip/files/5490465/adm-zip-crc32error.zip)

Zipinfo and unzip tools shows the zip file is ok.
Zip file created with [PHP ziparchive](https://www.php.net/manual/en/class.ziparchive.php).

Using "adm-zip": "0.4.16" within [binary-nwjs-v0.49.2-linux-x64](https://nwjs.io/blog/v0.49.2/).

Code sample:
```
var zip = new AdmZip(file);
if (zip) {
	try {
		zip.extractAllTo(sPath);
		console.log(sPath);
	} catch (e) {
		console.log(e.toString());
	}
}
```
**Error: CRC32 checksum failed 3c9180b82f59_MDFe31201000634453001737580010000035571000035578.pdf**

Edit: I have just tryed "adm-zip": "https://github.com/cthackers/adm-zip/tarball/master" and it did not worked as well.

**Comment by 5saviahv** [comment]
Interesting It worked well in my case

```
#!/usr/bin/env node

const AdmZip = require('adm-zip');

function extract(file, sPath){
    const zip = new AdmZip(file);
    if (zip) {
        try {
            zip.extractAllTo(sPath);
            console.log(sPath);
        } catch (e) {
            console.log(e.toString());
        }
    }
}

extract('./issue_330/adm-zip-crc32error.zip','./issue_330/unzipped');
```

... but when I think I used node v10.23.0

**Comment by EthraZa** [comment]
More interesting yet, I just tested it in the cmd with your code and it worked (node v15.0.1).
Another thing is that I used the "adm-zip-fixcrc": "^0.1.3" inside nwjs and it worked right.

So, looks like adm-zip is bugging to uncompress some files inside nwjs, that it can uncompress ok using just node in the cmd, but adm-zip-fixcrc is working right at both.

**Comment by 5saviahv** [comment]
I compared two libraries today and I didnt see big differences with fixcrc (fixcrc is fork after all). but adm-zip will "throw" with errors, when fixcrc handle them more gradually. Maybe problem is there. I also try setup nwjs for testing (I havent used nwjs before).

**Comment by 5saviahv** [comment]
This error seems to come from **zlib**. I modified code so it will give SHA1 hash values before and after extraction with (zlib.inflateRawSync) and Rrsults are:

NWJS Source: `d2aaec6e14bf9aca9f805f37eab0e5d1f0042ef4` Result: `c27789e1480426147283ec99d2b3e21f224c77b6`
Node Source: `d2aaec6e14bf9aca9f805f37eab0e5d1f0042ef4` Result: `1a152d782882a65a39249fd655bfa7ad92746097`

I also looked what went wrong and in NWJS output - there is like 98 byte long cap in data.

Node is 10.23.0 and in NWJS is node 15.0.1

I also discovered fixcrc does give same error, but it is shown in different location. If you right click on NWJS window, select "Inspect background page" and you see.

I really liked NWJS it seems to be easier make smaller apps than it is on Electron.
